login/src/main/java/com/fightpandemics/login/ui/Auth0BaseFragment.kt

```diff
@@ -39,54 +32,56 @@ open class Auth0BaseFragment(
         auth0.isOIDCConformant = true
     }
 
-    fun doSocialLogin(loginConnection: LoginConnection) {
-        WebAuthProvider.login(auth0)
-            .withConnection(loginConnection.provider)
-            .withScheme(SCHEME)
-            .withScope("openid profile email")
-            .withResponseType(ResponseType.CODE )
-            .withState("fight-pandemics")
-            .withAudience(
-                String.format(CALLBACK_START_URL, getString(R.string.com_auth0_domain))
-            )
-            .start(
-                requireActivity(),
-                Auth0CallBack(
-                    {
-                        Toast.makeText(
-                            context,
-                            "Unexpected error, try again later",
-                            Toast.LENGTH_LONG
-                        )
-                    },
-                    {
-                        auth0 = Auth0(requireContext())
-                        auth0.isOIDCConformant = true
-                        authTokenLocalDataSource.setToken(it.accessToken)
-
-                        val accessToken = it.accessToken
-                        if (accessToken != null) {
-                            getProfile(accessToken) {
-                                authTokenLocalDataSource.setUserId(it?.appMetadata?.get("mongo_id") as String?)
-                                when (loginConnection) {
-                                    LoginConnection.LINKEDIN_SIGNUP, LoginConnection.FACEBOOK_SIGNUP, LoginConnection.GOOGLE_SIGNUP -> {
-                                        findNavController().navigate(R.id.action_signUpFragment_to_completeProfileFragment)
-                                    }
-                                    LoginConnection.LINKEDIN_SIGNIN, LoginConnection.FACEBOOK_SIGNIN, LoginConnection.GOOGLE_SIGNIN -> {
-                                        goMain()
-                                    }
+    fun doSocialLogin(loginConnection: LoginConnection) = WebAuthProvider.login(auth0)
+        .withConnection(loginConnection.provider)
+        .withScheme(SCHEME)
+        .withScope(SCOPE)
+        .withResponseType(ResponseType.CODE)
+        .withState(STATE)
+        .withAudience(String.format(CALLBACK_PROFILE_URL, getString(R.string.com_auth0_domain)))
+        .start(
+            requireActivity(),
+            Auth0CallBack(
+                {
+                    Toast.makeText(
+                        context,
+                        "Unexpected error, try again later",
```

Comment:
Isn't it better to have this as a string resource?

Reply:
**yes I will do it**

Reply:
done!